src/hooks/useSessionWithWebApp.ts

```diff
@@ -0,0 +1,149 @@
+interface IWebAppParameterRef {
+	/** Optional id of the referenced parameter. Takes precedence over name. Id or name must be specified. */
+	id?: string
+	/** Optional name of the referenced parameter. Id takes precedence. Id or name must be specified. */
+	name?: string
+	/** Properties of the parameter to be overridden. */
+	overrides?: Pick<IWebAppParameterDefinition, "displayname" | "group" | "order" | "tooltip" | "hidden">
+	/** Disable the UI element of the parameter if its state is dirty. */
+	disableIfDirty: boolean
+	/** Ask the user to accept or reject changes of this parameter before executing them. */
+	acceptRejectMode: boolean
+}
+
+/** Reference to an export (defined by the session) */
+interface IWebAppExportRef {
+	/** Optional id of the referenced export. Takes precedence over name. Id or name must be specified. */
+	id?: string
+	/** Optional name of the referenced export. Id takes precedence. Id or name must be specified. */
+	name?: string
+	/** Properties of the export to be overridden. */
+	overrides?: Pick<IWebAppExportDefinition, "displayname" | "group" | "order" | "tooltip" | "hidden">
+}
+
+/** Types of widgets */
+type WebAppWidgetType = "accordion" | "text" | "image";
+
+/** 
+ * Properties of a parameter and export accordion widget.
+ * UI elements of the referenced parameters and exports are grouped 
+ * and ordered according to their properties (which might be overridden).
+ */
+interface IWebAppWidgetPropsAccordion {
+	/** References to parameters which shall be displayed by the accordion. */
+	parameters: IWebAppParameterRef[]
+	/** References to exports which shall be displayed by the accordion. */
+	exports: IWebAppExportRef[]
+}
+
+/** Properties of a text widget. */
+interface IWebAppWidgetPropsText {
+	/** Plain text. Takes precedence. */
+	text?: string
+	/** Optional markdown. */
+	markdown?: string
+}
+
+/** Properties of an image widget. */
+interface IWebAppWidgetPropsImage {
+	/** Base 64 encoded image. Takes precedence over export reference. */
+	data?: string
+	/** Optional reference to export which provides the image */
+	export?: IWebAppExportRef
+}
+
+/** A widget. */
+interface IWebAppWidget {
+	/** Type of the widget. */
+	type: WebAppWidgetType
+	/** Properties of the widget. */
+	props: IWebAppWidgetPropsAccordion | IWebAppWidgetPropsText | IWebAppWidgetPropsImage
+}
+
+/** 
+ * A tab displayed in a container.
+ */
+interface IWebAppContainerTab {
+	/** Name of the tab. */
+	name: string
+	/** Widgets displayed in the tab. */
+	widgets: IWebAppWidget[]
+}
+
+/** Types of hints for containers */
+type WebAppContainerHintType = "left" | "right" | "top" | "bottom";
+
+/**
+ * A container for UI elements
+ */
+interface IWebAppContainer {
+	/** Hint on positioning of the container. */
+	hint?: WebAppContainerHintType
+	/** Tabs displayed in the container. */
+	tabs?: IWebAppContainerTab[]
+	/** Further widgets displayed in the container. */
+	widgets?: IWebAppWidget[]
```

Comment:
@snabela So there can be widgets directly in a container, outside of the tabs?

Reply:
@mathieu-shapediver I thought this might be useful to display text or images, below the tabbed parameter panels.